src/Arcus.WebApi.Logging.Core/Correlation/HttpCorrelation.cs

```diff
@@ -115,45 +115,59 @@ private string DetermineOperationId(HttpContext httpContext)
         {
             if (String.IsNullOrWhiteSpace(httpContext.TraceIdentifier))
             {
+                _logger.LogTrace("No unique trace identifier ID was found in the request, generating one...");
                 string operationId = _options.Operation.GenerateId();
+                
                 if (String.IsNullOrWhiteSpace(operationId))
                 {
                     throw new InvalidOperationException(
                         $"Correlation cannot use '{nameof(_options.Operation.GenerateId)}' to generate an operation ID because the resulting ID value is blank");
                 }
 
+                _logger.LogTrace("Generated '{OperationId}' as unique operation correlation ID", operationId);
                 return operationId;
             }
 
+            _logger.LogTrace("Found unique trace identifier ID '{TraceIdentifier}' for operation correlation ID", httpContext.TraceIdentifier);
             return httpContext.TraceIdentifier;
         }
 
         private string DetermineTransactionId(StringValues transactionIds)
         {
-            if (String.IsNullOrWhiteSpace(transactionIds.ToString()))
+            var alreadyPresentTransactionId = transactionIds.ToString();
+            if (String.IsNullOrWhiteSpace(alreadyPresentTransactionId))
             {
                 if (_options.Transaction.GenerateWhenNotSpecified)
                 {
-                    string transactionId = _options.Transaction.GenerateId();
-                    if (String.IsNullOrWhiteSpace(transactionId))
+                    _logger.LogTrace("No transactional ID was found in the request, generating one...");
+                    string newlyGeneratedTransactionId = _options.Transaction.GenerateId();
+
+                    if (String.IsNullOrWhiteSpace(newlyGeneratedTransactionId))
                     {
                         throw new InvalidOperationException(
                             $"Correlation cannot use function '{nameof(_options.Transaction.GenerateId)}' to generate an transaction ID because the resulting ID value is blank");
                     }
 
-                    return transactionId;
+                    _logger.LogTrace("Generated '{TransactionId}' as transactional correlation ID", newlyGeneratedTransactionId);
+                    return newlyGeneratedTransactionId;
                 }
 
+                _logger.LogWarning(
```

Comment:
I'd go with a trace here as this can be noisy while they've turned it off as well

Reply:
Ok, changed. Resolving.